### ISSUE TITLE
Problems with backups because of LuceneIndexes lock

### ISSUE THREAD

**Issue by jfernandezrico** [orientechnologies/orientdb#5910]
I'm not able to run a backup because of lucene index files 
D:\orientdb-enterprise-2.1.11\databases\sftech\luceneIndexes\Ambito.ambito\write.lock

`- Compressing file luceneIndexes\Ambito.ambito\write.lock... [OAutomaticBackup$1$2]
2016-03-31 07:00:02:300 INFO  {db=sftech} error: java.io.IOException: El proceso no tiene acceso al archivo porque otro proceso tiene bloqueada una parte del archivo [OAutomaticBackup$1$2]{db=sftech} Cannot compress file: D:\orientdb-enterprise-2.1.11\databases\sftech\luceneIndexes\Ambito.ambito\write.lock
java.io.IOException: El proceso no tiene acceso al archivo porque otro proceso tiene bloqueada una parte del archivo
	at java.io.FileInputStream.readBytes(Native Method)
	at java.io.FileInputStream.read(FileInputStream.java:272)
	at com.orientechnologies.common.io.OIOUtils.copyStream(OIOUtils.java:170)
	at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUtil.addFile(OZIPCompressionUtil.java:186)
	at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUtil.addFolder(OZIPCompressionUtil.java:141)
	at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUtil.addFolder(OZIPCompressionUtil.java:127)
	at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUtil.addFolder(OZIPCompressionUtil.java:127)
	at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUtil.addFolder(OZIPCompressionUtil.java:127)
	at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUtil.compressDirectory(OZIPCompressionUtil.java:57)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.backup(OLocalPaginatedStorage.java:158)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.backup(ODatabaseDocumentTx.java:2816)
	at com.orientechnologies.orient.server.handler.OAutomaticBackup$1.run(OAutomaticBackup.java:161)
	at java.util.TimerThread.mainLoop(Timer.java:555)
	at java.util.TimerThread.run(Timer.java:505)
{db=sftech} [OAutomaticBackup] - Error on exporting database 'plocal:D:/orientdb-enterprise-2.1.11/databases/sftech' to file: backup/sftech-20160331070000.zip
java.io.IOException: El proceso no tiene acceso al archivo porque otro proceso tiene bloqueada una parte del archivo
	at java.io.FileInputStream.readBytes(Native Method)
	at java.io.FileInputStream.read(FileInputStream.java:272)
	at com.orientechnologies.common.io.OIOUtils.copyStream(OIOUtils.java:170)
	at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUtil.addFile(OZIPCompressionUtil.java:186)
	at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUtil.addFolder(OZIPCompressionUtil.java:141)
	at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUtil.addFolder(OZIPCompressionUtil.java:127)
	at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUtil.addFolder(OZIPCompressionUtil.java:127)
	at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUtil.addFolder(OZIPCompressionUtil.java:127)
	at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUtil.compressDirectory(OZIPCompressionUtil.java:57)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.backup(OLocalPaginatedStorage.java:158)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.backup(ODatabaseDocumentTx.java:2816)
	at com.orientechnologies.orient.server.handler.OAutomaticBackup$1.run(OAutomaticBackup.java:161)
	at java.util.TimerThread.mainLoop(Timer.java:555)
	at java.util.TimerThread.run(Timer.java:505)

2016-03-31 07:00:02:315 INFO  [OAutomaticBackup] Backup finished: 0 ok, 1 errors [OAutomaticBackup$1]`

How can I solve this?

Thank you in advance

**Comment by maggiolo00** [comment]
hi @jfernandezrico 

let me check.

**Comment by jfernandezrico** [comment]
Ok. I think I create the Lucene indexes well. I don't know if write.lock files have to be removed.

Thank you

**Comment by jfernandezrico** [comment]
Could you see anything about this?

**Comment by robfrank** [comment]
I need more details on how your doing the backup. The command, the kind of backup (plocal or remote), every detail that enable me to reproduce the problem. I guess you were using 2.1.11 from the stacktrace. Did you try with newer versions?

**Comment by jfernandezrico** [comment]
Hello robfrank. I just upgraded to 2.1.17 and I get the same error. I think it's because of the Lucene Indexes.
My backups are configurated within the orientdb-server-config.xml:

```
<handler class="com.orientechnologies.orient.server.handler.OAutomaticBackup">
            <parameters>
                <parameter value="true" name="enabled"/>
                <parameter value="4h" name="delay"/>
                <parameter value="11:35:00" name="firstTime"/>
                <parameter value="backup" name="target.directory"/>
                <parameter value="${DBNAME}-${DATE:yyyyMMddHHmmss}.zip" name="target.fileName"/>
                <parameter value="9" name="compressionLevel"/>
                <parameter value="1048576" name="bufferSize"/>
                <parameter value="sftech" name="db.include"/>
                <parameter value="" name="db.exclude"/>
            </parameters>
        </handler>
```


2016-05-17 11:35:05:302 INFO  {db=sftech} error: java.io.IOException: El proceso
 no tiene acceso al archivo porque otro proceso tiene bloqueada una parte del archivo [OAutomaticBackup$1$2]{db=sftech} Cannot compress file: D:\orientdb-enterp
rise-2.1.17\databases\sftech\luceneIndexes\Ambito.ambito\write.lock
java.io.IOException: El proceso no tiene acceso al archivo porque otro proceso t
iene bloqueada una parte del archivo
        at java.io.FileInputStream.readBytes(Native Method)
        at java.io.FileInputStream.read(FileInputStream.java:272)
        at com.orientechnologies.common.io.OIOUtils.copyStream(OIOUtils.java:170
)
        at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUti
l.addFile(OZIPCompressionUtil.java:186)
        at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUti
l.addFolder(OZIPCompressionUtil.java:141)
        at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUti
l.addFolder(OZIPCompressionUtil.java:127)
        at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUti
l.addFolder(OZIPCompressionUtil.java:127)
        at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUti
l.addFolder(OZIPCompressionUtil.java:127)
        at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUti
l.compressDirectory(OZIPCompressionUtil.java:57)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocal
PaginatedStorage.backup(OLocalPaginatedStorage.java:158)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.bac
kup(ODatabaseDocumentTx.java:2754)
        at com.orientechnologies.orient.server.handler.OAutomaticBackup$1.run(OA
utomaticBackup.java:161)
        at java.util.TimerThread.mainLoop(Timer.java:555)
        at java.util.TimerThread.run(Timer.java:505)
{db=sftech} [OAutomaticBackup] - Error on exporting database 'plocal:D:/orientdb
-enterprise-2.1.17/databases/sftech' to file: backup/sftech-20160517113500.zip
java.io.IOException: El proceso no tiene acceso al archivo porque otro proceso t
iene bloqueada una parte del archivo
        at java.io.FileInputStream.readBytes(Native Method)
        at java.io.FileInputStream.read(FileInputStream.java:272)
        at com.orientechnologies.common.io.OIOUtils.copyStream(OIOUtils.java:170
)
        at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUti
l.addFile(OZIPCompressionUtil.java:186)
        at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUti
l.addFolder(OZIPCompressionUtil.java:141)
        at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUti
l.addFolder(OZIPCompressionUtil.java:127)
        at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUti
l.addFolder(OZIPCompressionUtil.java:127)
        at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUti
l.addFolder(OZIPCompressionUtil.java:127)
        at com.orientechnologies.orient.core.compression.impl.OZIPCompressionUti
l.compressDirectory(OZIPCompressionUtil.java:57)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocal
PaginatedStorage.backup(OLocalPaginatedStorage.java:158)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.bac
kup(ODatabaseDocumentTx.java:2754)
        at com.orientechnologies.orient.server.handler.OAutomaticBackup$1.run(OA
utomaticBackup.java:161)
        at java.util.TimerThread.mainLoop(Timer.java:555)
        at java.util.TimerThread.run(Timer.java:505)

2016-05-17 11:35:05:349 INFO  [OAutomaticBackup] Backup finished: 0 ok, 1 errors
 [OAutomaticBackup$1]

**Comment by robfrank** [comment]
Hi, I can't replicate the problem, it seems related to windows. I don't have a windows box to test. I'll retry on windows in a few days.

**Comment by robfrank** [comment]
implemented freeze & release of lucene indexes. will be available on 2.2.1
